packages/scheduler/lib/workers/scheduling/scheduling.worker.ts

```diff
@@ -116,11 +116,16 @@ export class SchedulingChild {
                             createdToStartedTimeoutSecs: schedule.createdToStartedTimeoutSecs,
                             startedToCompletedTimeoutSecs: schedule.startedToCompletedTimeoutSecs,
                             heartbeatTimeoutSecs: schedule.heartbeatTimeoutSecs
-                        });
-                        if (task.isErr()) {
-                            logger.error(`Failed to create task for schedule: ${schedule.id}`);
+                        })
+                    );
+                    const res = await Promise.allSettled(createTasks);
+                    for (const taskRes of res) {
+                        if (taskRes.status === 'rejected') {
+                            logger.error(`Failed to schedule task: ${taskRes.reason}`);
+                        } else if (taskRes.value.isErr()) {
+                            logger.error(`Failed to schedule task: ${stringifyError(taskRes.value.error)}`);
```

Comment:
you could have done all of that inside the .map (if that makes more sense)

Reply:
you mean with a `then`? The goal is not to wait for the result in the map

Reply:
I meant keeping the code exactly the same but inside the map, since you don't kill the transaction
```ts
schedules.value.map(async (schedule) => {
      const task = await tasks.create(trx, {
          scheduleId: schedule.id,
          startsAfter: new Date(),
          name: `${schedule.name}:${new Date().toISOString()}`,
          createdToStartedTimeoutSecs: schedule.createdToStartedTimeoutSecs,
          startedToCompletedTimeoutSecs: schedule.startedToCompletedTimeoutSecs,
          heartbeatTimeoutSecs: schedule.heartbeatTimeoutSecs
      });
      if (task.isErr()) 
      [...]
  });

Reply:
I see.

Reply:
should not no, this is a common pattern `await Promise.all(array.map(async))`